javascript-modal/main.js

```diff
@@ -0,0 +1,25 @@
+var openModalButton = document.querySelector('.open-modal');
+var openModal = false;
+var overlayElement = document.querySelector('.overlay');
+var modalElement = document.querySelector('.modal');
+var modalButton = document.querySelector('.modal-button');
+var closeModal = false;
+
+openModalButton.addEventListener('click', modalHandle);
+modalButton.addEventListener('click', modalButtonHandle);
+
+function modalHandle(event) {
+  openModal = !openModal;
+  if (openModal) {
+    overlayElement.className = 'overlay';
+    modalElement.className = 'modal';
+  }
+}
+
+function modalButtonHandle(event) {
+  closeModal = !closeModal;
```

Comment:
Since you have this in multiple different functions, a boolean flag is not needed. As it is right now, for both of these functions, you only update the DOM if the variable value is `true`, which means every other time your event handler function runs, the classes aren't updated. You can make use of a boolean flag if you only made one event handler function, but with two separate function, you can just do the work without needing to check anything. If somebody clicks the "Open Modal" button, they want to open the modal every time. If somebody clicks the "No" button, they want to close the modal every time.